src/zelos/ext/plugins/yarascan/yarascan.py

```diff
@@ -0,0 +1,423 @@
+        if self._count_xrefs:
+            f.write(f"\txrefs: {self.xrefs}\n")
+        if not brief:
+            f.write(f"\tstrings:\n")
+            for i, s in enumerate(self.strings):
+                f.write(f"\t\taddress: 0x{self._mr.start + s.offset:x}\n")
+                f.write(f"\t\t\toffset: 0x{s.offset:x}\n")
+                f.write(f'\t\t\tvalue: "{s.value[:1000]}"\n')
+                if self._count_xrefs:
+                    f.write(f"\t\t\txrefs: {self._xref_cnts[i]}\n")
+        return f.getvalue()
+
+    def info(self, brief: bool = False) -> str:
+        """
+        Get match summary information.
+
+        Args:
+            brief: if True, omit rule string match details.
+
+        Returns:
+            A one-line informational string.
+        """
+        if brief:
+            return f"Matched rule: {self.namespace}.{self.rule}"
+        else:
+            for i, s in enumerate(self.strings):
+                xref_info = ""
+                if self._count_xrefs:
+                    xref_info = f" xrefs: {self._xref_cnts[i]}"
+                return (
+                    f"Matched {self.namespace}.{self.rule}"
+                    f"0x{s.address:08x} +0x{s.offset:x}{xref_info} "
+                    f"{s.value[:100]}"
+                )
+
+    def memdump(self, directory: str) -> str:
+        """
+        Dumps the match memory region data to the specified directory.
+
+        Args:
+            directory: folder to dump the region to.
+
+        Returns:
+            The full path of the dumped memory region file.
+        """
+        filename = os.path.join(
+            directory, f"PID{self.pid}_0x{self.region_address:x}.mem"
+        )
+        pathlib.Path(filename).parent.mkdir(parents=True, exist_ok=True)
+        with open(filename, "wb") as f:
+            f.write(self._mr.get_data())
+            return filename
+        return None
+
+
+class YaraScan(IPlugin):
+    """
+    YaraScan plugin scans memory using Yara rules.
+    """
+
+    def __init__(self, z: Zelos):
+        super().__init__(z)
+        self._rules = None
+        self._cmdline_rules = None
+        if (
+            z.config.yara_file is None
+            and z.config.yara_rule is None
+            and z.config.yara_file_glob is None
+        ):
+            return
+        self._yara = None
+        try:
+            import yara
+
+            self._yara = yara
+        except ModuleNotFoundError:
+            self._log(f"optional dependency `yara-python` not installed.")
+            self._log(f"yara rules will be IGNORED.")
+            self._log(f"try `pip install yara-python`.")
+            return
+        self._z = z
+
+        def closure() -> None:
+            self.compile(
+                files=z.config.yara_file,
+                rules=z.config.yara_rule,
+                glob_string=z.config.yara_file_glob,
+            )
+            self._log("scanning memory...")
+            all(
```

Comment:
Isn't all that all does check to see whether all values are true? Are you intending to use the result of all somewhere?